src/constraint/encoding.ts

```diff
@@ -140,6 +140,34 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.TYPE],
```

Comment:
Please also add a few generate tests to ensure this works correctly.  (A part of the reason you miss this bug is that you don't do generate test.)